### PR TITLE
feat(tag-appender): Enable loading scripts in a specific order with a…

### PR DESCRIPTION
**Pull request by hesher**
… flag in angular-widget

### CODE REVIEW

File: app/scripts/services/sequential-file-loader.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  /* @ngInject */
+  function SequentialFileLoader($q, tagAppender) {
+    this.loadFiles = function (filenames) {
+      return filenames.reduce(function (previousPromise, filename) {
+        return previousPromise.then(function () {
+          return tagAppender(filename, filename.split('.').reverse()[0]);
+        });
+      }, $q.when());
+    };
+  }
```
Comment (shahata) [inline app/scripts/services/sequential-file-loader.js]: this should be internal method of file loader service

File: test/spec/services/file-loader.spec.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  // instantiate service
+  var fileLoader, $rootScope, $q;
+  beforeEach(inject(function (_fileLoader_, _$rootScope_, _$q_) {
+    fileLoader = _fileLoader_;
+    $rootScope = _$rootScope_;
+    $q = _$q_;
+  }));
+
+  describe('when processing parralel script loading (non-array parameters)', function () {
+      it('should handle files and arrays correctly', inject(function (tagAppender, sequentialFileLoader) {
```
Comment (shahata) [inline test/spec/services/file-loader.spec.js]: where are you testing that when all promises are resolved, the returned promise is resolved?

File: test/spec/services/file-loader.spec.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+  // load the service's module
+  beforeEach(function () {
+    module('angularWidgetInternal');
+    module({
+      tagAppender: jasmine.createSpy('tagAppender').andCallFake(function (filename) {
+        return filename + '.done';
+      }),
+      sequentialFileLoader: {
+        loadFiles: jasmine.createSpy().andCallFake(function (arr) {
```
Comment (shahata) [inline test/spec/services/file-loader.spec.js]: why do we need to mock sequentialFileLoader? you should only mock tagAppender

File: test/spec/services/sequential-file-loader.spec.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+      files.forEach(function (file) {
+        testForFile(file, isResolvedFn);
+      });
+
+      expect(tagAppender.callCount).toBe(0);
+    });
+  }
+
+  it('should load files sequentially', function () {
```
Comment (shahata) [inline test/spec/services/sequential-file-loader.spec.js]: where in this test are you testing that anything happens sequentially?

File: test/spec/services/sequential-file-loader.spec.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    var files = ['baroom.aaa', 'garoom.bbb', 'chicky.xxx'].map(toFileObj);
+    var resolved = false;
+
+    setupTagAppender(files);
+    sequentialFileLoader.loadFiles(files.map(toFileName)).then(function () {
+      resolved = true;
```
Comment (shahata) [inline test/spec/services/sequential-file-loader.spec.js]: why not use a spy?

File: test/spec/services/sequential-file-loader.spec.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    inject(function (tagAppender) {
+      $rootScope.$digest();
+
+      files.forEach(function (file) {
+        testForFile(file, isResolvedFn);
+      });
+
+      expect(tagAppender.callCount).toBe(0);
+    });
+  }
```
Comment (shahata) [inline test/spec/services/sequential-file-loader.spec.js]: all of the setup above seems overly complex and uncalled for. is this TDD?

**Comment by shahata** [comment]
Implementation looks good, but please merge the two services (and specs) into one.
Tests I have to say do not look TDD to me and therefore they do not test a lot of the functionality and mock too much...

**Comment by hesher** [comment]
Uploaded fixes

**Pull request closed without merging**